test/conf_include_test.c

```diff
@@ -0,0 +1,189 @@
+/*
+ * Copyright 2016-2018 The OpenSSL Project Authors. All Rights Reserved.
+ *
+ * Licensed under the OpenSSL license (the "License").  You may not use
+ * this file except in compliance with the License.  You can obtain a copy
+ * in the file LICENSE in the source distribution or at
+ * https://www.openssl.org/source/license.html
+ */
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <openssl/conf.h>
+#include <openssl/err.h>
+#include <openssl/ssl.h>
+#include <string.h>
+#include "testutil.h"
+
+#ifdef _WIN32
+# include <direct.h>
+# define DIRSEP '\\'
+# define CHDIR _chdir
+#else
+# include <unistd.h>
+# define DIRSEP '/'
+# define CHDIR chdir
+#endif
+
+/* changes path to that of the filename */
+static int change_path(const char *file)
+{
+    char *s = OPENSSL_strdup(file);
+    char *p;
+    int ret;
+
+    if (s == NULL)
+        return -1;
+
+    p = strrchr(s, DIRSEP);
```

Comment:
As I suspected, test didn't work in wine-assisted mingw cross-compile build :-( A lot of strange words, huh? But allow me to break it down. "mingw cross-compile build" means `Configure mingw64 --cross-compile-prefix=x86_64-w64-mingw32-`, which generates normal Windows binaries. This works/is supported on Linux and Cygwin. [In MSYS2 case it's also effectively cross-compile, but you don't use --cross-compile-prefix.] And "wine-assisted" mean that you can actually execute normal Windows binaries on your [x86] Linux system with [wine](https://www.winehq.org/)'s assistance. You can configure `wine` to be used transparently, so that you can simply invoke Windows programs directly from your shell prompt as if they were Linux programs. Or you can assign `wine` to EXE_SHELL environment variable prior executing OpenSSL's `make test`. Now, keyword is that command line arguments are passed verbatim, and what would Linux or Cygwin pass down to Windows program? Yes, slash-delimited path names. One can argue that this is a corner case, and one should simply not execute the test in such case, but here is the thing. Windows run-time actually doesn't make much of a difference between slash and backslash. You can refer to `c:\some\where.txt`, `c:/some/where.txt` and even `c:\some/where.txt` with *equal* success. And with this in mind one can as well argue that test program could as well check for *either*. It would make test work. BTW, we do run wine-assisted tests on Travis CI as part of opt-in extended tests or upon actual commit to repository.